asreview/hdf5_logging.py

```diff
@@ -0,0 +1,259 @@
+
+        for key, value in self.f.items():
+            if isinstance(self.f[key], h5py.Dataset):
+                if value.shape is not None:
+                    hdf_dict[key] = np.array(value)
+
+        hdf_dict["results"] = []
+        for i in itertools.count():
+            if str(i) not in self.f["results"]:
+                break
+            hdf_dict["results"].append(
+                _dict_from_group(self.f["results"][str(i)]))
+
+        return hdf_dict
+
+    def is_empty(self):
+        return len(self.f["results"]) == 0
+
+    def set_labels(self, y):
+        if "labels" not in self.f:
+            self.f.create_dataset("labels", y.shape, dtype=np.int, data=y)
+        else:
+            self.f["labels"][...] = y
+
+    def set_final_labels(self, y):
+        if "final_labels" not in self.f:
+            self.f.create_dataset(
+                "final_labels", y.shape, dtype=np.int, data=y)
+        else:
+            self.f["final_labels"][...] = y
+
+    def add_classification(self, idx, labels, methods, query_i):
+        """Add training indices and their labels.
+
+        Arguments
+        ---------
+        indices: list, np.array
+            A list of indices used for training.
+        labels: list
+            A list of labels corresponding with the training indices.
+        i: int
+            The query number.
+        """
+        g = _result_group(self.f, query_i)
+        if "new_labels" not in g:
+            g.create_group("new_labels")
+
+        g = g['new_labels']
+
+        np_methods = np.array(list(map(np.string_, methods)))
+        _append_to_dataset('idx', idx, g, dtype=np.int)
+        _append_to_dataset('labels', labels, g, dtype=np.int)
+        _append_to_dataset('methods', np_methods, g, dtype='S20')
+
+#         bool_array = np.full(self.n_labels(), False, dtype=bool)
+#         bool_array[idx] = True
+#         self.f["labels"][bool_array] = labels
+#         for i, index in enumerate(idx):
+#             self.f["labels"][index] = labels[i]
+
+    def add_proba(self, pool_idx, train_idx, proba, query_i):
+        """Add inverse pool indices and their labels.
+
+        Arguments
+        ---------
+        indices: list, np.array
+            A list of indices used for unlabeled pool.
+        pred: np.array
+            Array of prediction probabilities for unlabeled pool.
+        i: int
+            The query number.
+        """
+        g = _result_group(self.f, query_i)
+        g.create_dataset("pool_idx", data=pool_idx, dtype=np.int)
+        g.create_dataset("train_idx", data=train_idx, dtype=np.int)
+        g.create_dataset("proba", data=proba, dtype=np.float)
+
+    def add_settings(self, settings):
+        self.settings = settings
+        self.f.attrs.pop('settings', None)
+        self.f.attrs['settings'] = np.string_(json.dumps(vars(settings)))
+
+    def n_queries(self):
+        return len(self.f['results'].keys())
+
+    def review_state(self):
```

Comment:
I'm not sure what this is doing. A docstring would be nice. Is this a private method?

Reply:
No it is used by the review class to return to a proper state from a log file.